src/game/Coordinate.java

```diff
@@ -34,23 +33,62 @@ public int getX() {
 	public int getY() {
 		return this.y;
 	}
-	
+
 	public boolean isValid() {
 		return this.valid;
 	}
 
+	public Coordinate down() {
+		return new Coordinate(x, y + 1);
+	}
+	
+	public Coordinate up() {
+		return new Coordinate(x, y - 1);
+	}
+	
+	public Coordinate left() {
+		return new Coordinate(x - 1, y);
+	}
+	
+	public Coordinate right() {
+		return new Coordinate(x + 1, y);
+	}
```

Comment:
No longer needed. A new MoveCommand object handles these in a more abstract way.